pkg/dependency/parser/gradle/lockfile/parse.go

```diff
@@ -45,6 +45,10 @@ func (Parser) Parse(r dio.ReadSeekerAt) ([]types.Library, []types.Dependency, er
 					EndLine:   lineNum,
 				},
 			},
+			// There is no reliable way to determine direct dependencies (even using other files).
+			// Therefore, we mark all dependencies as Indirect.
+			// This is necessary to try to guess direct dependencies and build a dependency tree.
+			Indirect: true,
```

Comment:
I see a second way:
we can mark Gradle dependencies as indirect when before building the dependency tree - https://github.com/aquasecurity/trivy/blob/82214736a943f61c173902808f2887a660543fe2/pkg/report/table/vulnerability.go#L210